ClinicWeb/src/main/java/pl/infoshare/clinicweb/doctor/DoctorService.java

```diff
@@ -1,186 +1,190 @@
 package pl.infoshare.clinicweb.doctor;
 
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.repository.query.FluentQuery;
 import org.springframework.stereotype.Service;
-import pl.infoshare.clinicweb.file.FileService;
-import pl.infoshare.clinicweb.patient.Address;
-import pl.infoshare.clinicweb.user.PersonDetails;
-import pl.infoshare.clinicweb.user.User;
-
-import java.io.FileWriter;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Iterator;
+
 import java.util.List;
-import java.util.stream.Collectors;
+import java.util.Optional;
+import java.util.function.Function;
 
 @Service
 public class DoctorService implements DoctorRepository {
```

Comment:
```suggestion
public class DoctorService  {
```
zamiast tego robisz wstrzykiwanie przez konstruktor, dzieki temu będziesz miał dostęp do metod z repositorium jpa. 
Podmień od razu w odpowiednich metodach, 

private final DoctorRepository repository;